src/test/studio/kdb/KTest.java

```diff
@@ -104,4 +104,312 @@ public void testSymbolVector() {
+        check(new K.KDatetime(-123456.789), "1661.12.26 05:03:50.401", "1661.12.26 05:03:50.401z");
+        check(new K.KDatetime(123.456), "2000.05.03 10:56:38.400", "2000.05.03 10:56:38.400z");
+        check(new K.KDatetime(Double.NEGATIVE_INFINITY), "-0W", "-0Wz");
+        check(new K.KDatetime(Double.POSITIVE_INFINITY), "0W", "0Wz");
+        check(new K.KDatetime(Double.NaN), "0N", "0Nz");
+
+        check(vector(K.KDatetimeVector.class, -10.0, 10.0, 3.0), "1999.12.22 00:00:00.000 2000.01.11 00:00:00.000 2000.01.04 00:00:00.000", "1999.12.22 00:00:00.000 2000.01.11 00:00:00.000 2000.01.04 00:00:00.000z");
+        check(vector(K.KDatetimeVector.class), "`datetime$()", "`datetime$()");
+        check(vector(K.KDatetimeVector.class, 0.0), "enlist 2000.01.01 00:00:00.000", "enlist 2000.01.01 00:00:00.000z");
+        check(vector(K.KDatetimeVector.class, 5.0, Double.NEGATIVE_INFINITY, Double.POSITIVE_INFINITY, Double.NaN), "2000.01.06 00:00:00.000 -0W 0W 0N", "2000.01.06 00:00:00.000 -0W 0W 0Nz");
+    }
+
+
+    @Test
+    public void testListToString() throws Exception {
+        check(vector(K.KList.class), "()", "()");
+        check(vector(K.KList.class, new K.KLong(10), new K.KLong(Long.MAX_VALUE)), "(10;0W)", "(10j;0Wj)");
+        check(vector(K.KList.class, new K.KLong(10), new K.KInteger(10)), "(10;10)", "(10j;10i)");
+        check(vector(K.KList.class, new K.KLong(10), new K.KInteger(10), vector(K.KList.class, new K.KDouble(1.1))), "(10;10;enlist 1.1)", "(10j;10i;enlist 1.1f)");
+    }
+
+
+    @Test
+    public void testOtherToString() throws Exception {
+        K.Function funcUnary = new K.Function(new K.KCharacterVector("{1+x}"));
+        K.Function funcBinary = new K.Function(new K.KCharacterVector("{x+y}"));
+
+        check(funcUnary,"{1+x}", "{1+x}");
+        check(funcBinary,"{x+y}", "{x+y}");
+
+        check(new K.FEachLeft(funcBinary), "{x+y}\\:", "{x+y}\\:");
+        check(new K.FEachRight(funcBinary), "{x+y}/:", "{x+y}/:");
+        check(new K.Feach(funcBinary), "{x+y}'", "{x+y}'");
+        check(new K.Fover(funcBinary), "{x+y}/", "{x+y}/");
+        check(new K.Fscan(funcBinary), "{x+y}\\", "{x+y}\\");
+        check(new K.FPrior(funcBinary), "{x+y}':", "{x+y}':");
+
+        check(new K.FComposition(new Object[] {funcUnary, funcBinary}), "","");
+        check(new K.Projection((K.KList)vector(K.KList.class, funcBinary, new K.KLong(1), new K.UnaryPrimitive(-1))), "{x+y}[1;]", "{x+y}[1j;]");
+        check(new K.Projection((K.KList)vector(K.KList.class, funcBinary, new K.UnaryPrimitive(-1), new K.KLong(1))), "{x+y}[;1]", "{x+y}[;1j]");
+
+        check(new K.BinaryPrimitive(15), "~", "~");
+        check(new K.UnaryPrimitive(0), "::", "::");
+        check(new K.UnaryPrimitive(41), "enlist", "enlist");
+
+        //the output from +1
+        check(new K.Projection((K.KList)vector(K.KList.class, new K.BinaryPrimitive(1), new K.KLong(1))), "+[1]", "+[1j]");
+        //output from '[;]
```

Comment:
not sure what should be the output, orginal PR had `'[;]`, but origin code for kdb studio outputs `(';;)`

Reply:
similar comment for the 2 others "projections"

Reply:
The two have different meanings. ```'[;]``` is a projection of the ' operator with two missing arguments. ```(';;)``` is a projection of ```enlist``` with the first argument being ' and the second and third arguments elided. The first one seems to be more accurate.

Reply:
Fixed